src/test/java/lotto/model/LottoPurchaserTest.java

```diff
@@ -3,22 +3,30 @@
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 
+import java.util.List;
+
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class LottoPurchaserTest {
 
     @Test
-    @DisplayName("로또 금액이 들어가면 몇개의 로또를 사는지 테스트")
-    void 금액만큼_로또_구매() {
+    @DisplayName("자동 로또 정보와 수동 로또 정보가 주어지면 개수에 맞게 로또를 구매한다.")
+    void 자동_수동_로또_구매() {
         // given
-        LottoPurchaser purchaser = new LottoPurchaser(new AutoLottoGenerator());
+        LottoPurchaser purchaser = new LottoPurchaser(new LottoFactory());
+
+        Automatic automatic = new Automatic(new LottoMoney(14000));
+        Manuals manuals = new Manuals(List.of(
+                List.of(1, 2, 3, 4, 5, 6)), new LottoMoney(new ManualLottoCount(1)));
 
         // when
-        PurChasedLottos purchaseLotto = purchaser.purchaseLotto(new LottoMoney(14000));
+        PurChasedLottos purchaseLotto = purchaser.purchaseLotto(automatic, manuals);
 
         // then
-        assertThat(purchaseLotto.getLottoCount()).isEqualTo(14);
+        assertThat(purchaseLotto.getLottos().size()).isEqualTo(15);
+        assertThat(purchaseLotto.getAutoCount()).isEqualTo(14);
+        assertThat(purchaseLotto.getManualCount()).isEqualTo(1);
+        assertThat(purchaseLotto.getMoney()).isEqualTo(15000);
```

Comment:
```suggestion
        assertThat(purchaseLotto.getLottos().size()).isEqualTo(15);
        assertThat(purchaseLotto)
                .extracting(PurChasedLottos::getAutoCount, PurChasedLottos::getManualCount, PurChasedLottos::getMoney)
                .contains(14, 1, 15000);
```

`extracting`이라는 기능도 있으니 한번 활용해보아도 좋겠네요

Reply:
객체를 검증할 때 사용하면 아주 유용할 것 같습니다. 감사합니다!